File: brainbox/score_string.py
def score_of_string(s: str) -> int:
    total = 0
    for i in range(len(s) - 1):
        total += abs(ord(s[i]) - ord(s[i + 1]))
    return total


def test_score_of_string():
    # Provided example
    assert score_of_string("hello") == 13

    # Basic cases
    assert score_of_string("abc") == 2      # |97-98| + |98-99| = 1 + 1
    assert score_of_string("aaa") == 0      # all same letters
    assert score_of_string("zaz") == 50     # |122-97| + |97-122| = 25 + 25

    # Edge cases
    assert score_of_string("") == 0
    assert score_of_string("a") == 0

    print("All tests passed!")

test_score_of_string()

File: brainbox/score_string.ts
export function scoreOfString(s: string): number {
    let total = 0;
    for (let i = 0; i < s.length - 1; i++) {
        total += Math.abs(s.charCodeAt(i) - s.charCodeAt(i + 1));
    }
    return total;
}


import { scoreOfString } from "./score_string";

function testScoreOfString() {
    console.assert(scoreOfString("hello") === 13, "Test 1 failed");
    console.assert(scoreOfString("abc") === 2, "Test 2 failed");
    console.assert(scoreOfString("aaa") === 0, "Test 3 failed");
    console.assert(scoreOfString("zaz") === 50, "Test 4 failed");
    console.assert(scoreOfString("") === 0, "Test 5 failed");
    console.assert(scoreOfString("a") === 0, "Test 6 failed");

    console.log("All tests passed!");
}

testScoreOfString();
